validator/server_api/json.go

```diff
@@ -1,13 +1,173 @@
 package server_api
 
 import (
+	"bytes"
 	"encoding/base64"
+	"fmt"
+	"io"
 
 	"github.com/ethereum/go-ethereum/common"
 
 	"github.com/offchainlabs/nitro/validator"
 )
 
+type PreimagesMapJson struct {
+	Map map[common.Hash][]byte
+}
+
+func (m *PreimagesMapJson) MarshalJSON() ([]byte, error) {
+	size := 2 // {}
+	encoding := base64.StdEncoding
+	for key, value := range m.Map {
+		size += 5 // "":""
+		size += encoding.EncodedLen(len(key))
```

Comment:
I've moved the calculation out of the loop, but I don't think we need to pull `encoding.EncodedLen(32)` out into a global because it's just `(n + 2) / 3 * 4`